code/game/machinery/doors/multi_tile.dm

```diff
@@ -299,6 +299,82 @@
 		open(1)
 		lock(TRUE)
 
+/obj/structure/machinery/door/airlock/multi_tile/almayer/dropshiprear/handle_multidoor(old_dir, new_dir)
+	if(!(width > 1)) return //Bubblewrap
+	if(old_dir != new_dir)
+		var/adjust = width - 1
+		switch(old_dir)
+			if(NORTH)
+				switch(new_dir)
+					if(EAST)
+						Move(locate(x, y - adjust, z), dir)
+					if(WEST)
+						Move(locate(x, y, z), dir)
+					if(SOUTH)
+						Move(locate(x - adjust, y, z), dir)
+			if(EAST)
+				switch(new_dir)
+					if(NORTH)
+						Move(locate(x - adjust, y, z), dir)
+					if(WEST)
+						Move(locate(x, y - adjust, z), dir)
+					if(SOUTH)
+						Move(locate(x, y, z), dir)
+			if(WEST)
+				switch(new_dir)
+					if(NORTH)
+						Move(locate(x - adjust, y, z), dir)
+					if(EAST)
+						Move(locate(x, y - adjust, z), dir)
+					if(SOUTH)
+						Move(locate(x - adjust, y, z), dir)
+			if(SOUTH)
+				switch(new_dir)
+					if(NORTH)
+						Move(locate(x - adjust, y, z), dir)
+					if(EAST)
+						Move(locate(x, y, z), dir)
+					if(WEST)
+						Move(locate(x, y - adjust, z), dir)
+
+	update_filler_turfs()
+	if(dir in list(NORTH, SOUTH))
+		bound_height = world.icon_size * width
+		bound_width = world.icon_size
+	else if(dir in list(EAST, WEST))
+		bound_width = world.icon_size * width
+		bound_height = world.icon_size
+
+//We have to find these again since these doors are used on shuttles a lot so the turfs changes
+/obj/structure/machinery/door/airlock/multi_tile/almayer/dropshiprear/update_filler_turfs()
+	for(var/turf/T in multi_filler)
+		T.set_opacity(null)
+
+	multi_filler = list()
+	for(var/turf/T in get_filler_turfs())
+		T.set_opacity(opacity)
+		multi_filler += list(T)
+
+/obj/structure/machinery/door/airlock/multi_tile/almayer/dropshiprear/get_filler_turfs()
+	. = list()
+	for(var/i = 1, i < width, i++)
```

Comment:
```suggestion
	for(var/i in 1 to width - 1)
```

https://github.com/cmss13-devs/cmss13/blob/master/.github/guides/STANDARDS.md#forvara-in-list-versus-forvari-in-1-to-listlen